apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -181,20 +182,20 @@ func NewProvisionerAPI(ctx facade.Context) (*ProvisionerAPI, error) {
 			serviceFactory.ControllerConfig(),
 			serviceFactory.ExternalController(),
 		),
-		NetworkConfigAPI:        netConfigAPI,
-		st:                      st,
-		m:                       model,
-		controllerConfigService: serviceFactory.ControllerConfig(),
-		resources:               resources,
-		authorizer:              authorizer,
-		configGetter:            configGetter,
-		storageProviderRegistry: storageProviderRegistry,
-		storagePoolManager:      poolmanager.New(state.NewStateSettings(st), storageProviderRegistry),
-		getAuthFunc:             getAuthFunc,
-		getCanModify:            getCanModify,
-		providerCallContext:     callCtx,
-		controllerUUID:          controllerCfg.ControllerUUID(),
-		logger:                  ctx.Logger().Child("provisioner"),
+		NetworkConfigAPI:                netConfigAPI,
+		st:                              st,
+		m:                               model,
+		controllerConfigService:         serviceFactory.ControllerConfig(),
+		resources:                       resources,
+		authorizer:                      authorizer,
+		configGetter:                    configGetter,
+		storageProviderRegistry:         storageProviderRegistry,
+		storagePoolManager:              poolmanager.New(state.NewStateSettings(st), storageProviderRegistry),
+		getAuthFunc:                     getAuthFunc,
+		getCanModify:                    getCanModify,
+		credentialInvalidatorFuncGetter: credentialcommon.CredentialInvalidatorFuncGetter(ctx),
```

Comment:
CredentialInvalidatorFuncGetter 🤮

Not sure how we got `InvalidateModelCredentialFuncGetter` and `CredentialInvalidatorFuncGetter`

They are both equally horrible. `ModelCredentialInvalidatorGetter` and `CredentialInvalidatorGetter` feel like much better names, but I'll leave this up to you.

Reply:
I hate naming things. Happy to use those other ones.